$OpenBSD: patch-src_couchdb_priv_couch_js_http_c,v 1.1 2011/04/10 17:28:38 jasper Exp $

"Starting in SpiderMonkey 1.8.5, jsval can store a full 32-bit integer, so this
check isn't needed any longer for 32-bit integers."

--- src/couchdb/priv/couch_js/http.c.orig	Fri Apr  1 09:35:22 2011
+++ src/couchdb/priv/couch_js/http.c	Fri Apr  1 09:37:23 2011
@@ -286,16 +286,8 @@ status(JSContext* cx, JSObject* obj, jsval idval, jsva
         return JS_FALSE;
     }
     
-    if(INT_FITS_IN_JSVAL(http->last_status))
-    {
-        *vp = INT_TO_JSVAL(http->last_status);
-        return JS_TRUE;
-    }
-    else
-    {
-        JS_ReportError(cx, "INTERNAL: Invalid last_status");
-        return JS_FALSE;
-    }
+    *vp = INT_TO_JSVAL(http->last_status);
+    return JS_TRUE;
 }
 
 JSClass CouchHTTPClass = {
